### ISSUE TITLE
Update the API by to plugins to expose the ability to abort a batch

### ISSUE THREAD

**Issue by andsel** [elastic/logstash#14987]
## Context
With PR #14940 was introduced the exception `org.logstash.execution.AbortedBatchException` that can be used by plugins to request  Logstash to negatively ACK the batch they are processing. This is usually leveraged by plugins that have a retry loop to process data, for example `logstash-output-elasticsearch` (https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1119), that needs to be broken when the shutdown of the pipeline is requested.
Negatively ACK-ing a batch that's spooled from **persistent queue**, in case of a pipeline restart, means that the batch is reprocessed again on the next pipeline processing cycle.

## Change requested
To make more explicit the intention of the exception, this issue requests to modify the existing `WorkerLoop` to catch the exception but before the signature of some methods needs to be updated:
- https://github.com/elastic/logstash/blob/137168d1c60ec45df851712ad2f21242e2a0b2da/logstash-core/src/main/java/org/logstash/config/ir/CompiledPipeline.java#L361
- https://github.com/elastic/logstash/blob/137168d1c60ec45df851712ad2f21242e2a0b2da/logstash-core/src/main/java/org/logstash/config/ir/compiler/Dataset.java#L67
- the `DatasetCompiler` which generates the code that wraps the invocation of plugins
- the various input|filter|output delegator's methods that could raise, like for example the `multiReceive` in the output delegator: https://github.com/elastic/logstash/blob/137168d1c60ec45df851712ad2f21242e2a0b2da/logstash-core/src/main/java/org/logstash/config/ir/compiler/AbstractOutputDelegatorExt.java#L116

**Comment by robbavey** [comment]
@andsel Is this still relevant/required?

**Comment by andsel** [comment]
Yes, we still need to update the signature of methods, and maybe we could also rename the exception, I would suggest `RejectedBatchException`, could be more appropriate.
